src/main/scala/tech/cryptonomic/conseil/tezos/michelson/parser/JsonParser.scala

```diff
@@ -153,11 +153,32 @@ object JsonParser {
     }
   }
 
-  /* Parses Michelson Expression represented as JSON to domain objects */
-  def parse(json: String): Result[MichelsonSchema] = {
+  trait Parser[T <: MichelsonElement] {
```

Comment:
Q: just to understand, is this used along the lines of how circe's parsers work, or is totally unrelated?

Reply:
It's unrelated. It's just an API for our Parser which hides what library we use.

Reply:
ty